api/src/main/java/org/apache/cloudstack/api/command/admin/vm/MigrateVMCmd.java

```diff
@@ -192,4 +187,25 @@ public void execute() {
             throw new ServerApiException(ApiErrorCode.INTERNAL_ERROR, e.getMessage());
         }
     }
+
+    @Override
+    public String getSyncObjType() {
```

Comment:
Why did you need to implemente these `getSyncObjType` and `getSyncObjId`? Could you explain this a little bit further?

Reply:
i can, but it will be not a little bit ;)
tl;dr: limitting the number of jobs send to vmware.
There is a mechanism to limit jobs. It checks in the sync phase of async jobs to see if the queue for those jobs exeeds a certain length. If so it return to the user to try again later. It can filter on type and entity, if it can get them. These methods are part of those mechs.